File: src/main/java/com/rafaelfiume/tictactoe/console/BlockingConsoleInputReader.java
package com.rafaelfiume.tictactoe.console;

import com.rafaelfiume.tictactoe.BoardPosition;

import java.util.Scanner;

public class BlockingConsoleInputReader implements ConsoleInputReader {

    private final Scanner scanner = new Scanner(System.in);

    @Override
    public BoardPosition readUserInput() {
        return BoardPosition.of(scanner.nextInt());
    }

}

File: src/main/java/com/rafaelfiume/tictactoe/console/BlockingConsoleInputReader.kt
package com.rafaelfiume.tictactoe.console

import com.rafaelfiume.tictactoe.BoardPosition

import java.util.Scanner

class BlockingConsoleInputReader : ConsoleInputReader {

    private val scanner = Scanner(System.`in`)

    override fun readUserInput(): BoardPosition = BoardPosition.of(scanner.nextInt())

}
